fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java

```diff
@@ -6966,6 +6993,124 @@ public LogicalPlan visitTransactionRollback(DorisParser.TransactionRollbackConte
         return new TransactionRollbackCommand();
     }
 
+    @Override
+    public LogicalPlan visitGrantTablePrivilege(DorisParser.GrantTablePrivilegeContext ctx) {
+        List<AccessPrivilegeWithCols> accessPrivilegeWithCols = visitPrivilegeList(ctx.privilegeList());
+
+        List<String> parts = visitMultipartIdentifierOrAsterisk(ctx.multipartIdentifierOrAsterisk());
+        int size = parts.size();
+
+        if (size < 1) {
+            throw new AnalysisException("grant table privilege statement missing parameters");
+        }
+
+        TablePattern tablePattern = null;
+        if (size == 1) {
+            String db = parts.get(size - 1);
+            tablePattern = new TablePattern(db, "");
+        }
+
+        if (size == 2) {
+            String db = parts.get(size - 2);
+            String tbl = parts.get(size - 1);
+            tablePattern = new TablePattern(db, tbl);
+        }
+
+        if (size == 3) {
+            String ctl = parts.get(size - 3);
+            String db = parts.get(size - 2);
+            String tbl = parts.get(size - 1);
+            tablePattern = new TablePattern(ctl, db, tbl);
+        }
+
+        Optional<UserIdentity> userIdentity = ctx.userIdentify() == null
+                ? Optional.empty() : Optional.of(visitUserIdentify(ctx.userIdentify()));
+        Optional<String> role = ctx.ROLE() == null
+                ? Optional.empty() : Optional.of(visitIdentifierOrText(ctx.identifierOrText()));
+
+        return new GrantTablePrivilegeCommand(
+            accessPrivilegeWithCols,
+            tablePattern,
+            userIdentity,
+            role);
+    }
+
+    @Override
+    public LogicalPlan visitGrantResourcePrivilege(DorisParser.GrantResourcePrivilegeContext ctx) {
+        List<AccessPrivilegeWithCols> accessPrivilegeWithCols = visitPrivilegeList(ctx.privilegeList());
+
+        String name = visitIdentifierOrTextOrAsterisk(ctx.identifierOrTextOrAsterisk());
+        ResourcePattern resourcePattern = null;
+        WorkloadGroupPattern workloadGroupPattern = null;
+
+        if (ctx.CLUSTER() != null || ctx.COMPUTE() != null) {
+            resourcePattern = new ResourcePattern(name, ResourceTypeEnum.CLUSTER);
+        } else if (ctx.STAGE() != null) {
+            resourcePattern = new ResourcePattern(name, ResourceTypeEnum.STAGE);
+        } else if (ctx.STORAGE() != null) {
+            resourcePattern = new ResourcePattern(name, ResourceTypeEnum.STORAGE_VAULT);
+        } else if (ctx.RESOURCE() != null) {
+            resourcePattern = new ResourcePattern(name, ResourceTypeEnum.GENERAL);
+        }
+
+        if (ctx.WORKLOAD() != null) {
+            workloadGroupPattern = new WorkloadGroupPattern(name);
```

Comment:
for workload group, it's resource type should always be  ResourceTypeEnum.GENERAL